### ISSUE TITLE
Setting map center to L.LatLng(Infinity, Infinity) causes infinite loop

### ISSUE THREAD

**Issue by theashyster** [Leaflet/Leaflet#5477]
## How to reproduce
- Leaflet version I'm using: `master`, 1.0.3
- Browser (with version) I'm using: Chrome, Firefox (latest versions)
- OS/Platform (with version) I'm using: Windows 8.1

## What behavior I'm expecting and which behavior I'm seeing
I expect that passing `Infinity` in to the `L.LatLng` constructor like this `new L.LatLng(Infinity, Infinity);` would throw an error. Instead this is allowed and is causing the browser to freeze when adding a tile layer, for example.

## Minimal example reproducing the issue
**Beware, this will freeze your browser**
Example - http://playground-leaflet.rhcloud.com/qem/edit?html,output

**Comment by IvanSanchez** [comment]
The problem, as far as I can see, lies within the `L.GridLayer` logic for the tile bounds. The `for (var i = tileRange.min.x; i <= tileRange.max.x; i++) {` loop in `L.GridLayer._update` is trying to loop from `Infinity` to `Infinity` (since those are the values of `tileRange.min.x` and `tileRange.max.x`)

**Comment by theashyster** [comment]
I see, okay, then we should blame the `L.GridLayer`.
Yep, great. One the other hand though - should the `L.LatLng` be initiated using infinite numbers?

**Comment by IvanSanchez** [comment]
I think that `L.LatLng`s should be allowed to have `Infinity` coordinates. We, in fact, use stuff like that for the `maxBounds` option of the map.

**Comment by theashyster** [comment]
Okay, I see, thanks for the explanation. ;)

**Comment by theashyster** [comment]
Looks like this is fixed in #5479

**Comment by theashyster** [comment]
@IvanSanchez However, looks like I found another issue after the merge.
This is the same example, but with the Console part open - http://playground-leaflet.rhcloud.com/qem/edit?html,console,output

The map loads up and if you do a drag on the map the Console will continuously fill up with the newly added error `Attempted to load an infinite number of tiles`.

**Comment by IvanSanchez** [comment]
@theashyster Hey, you wanted Leaflet to throw an error instead of getting stuck in an infinite loop, and you got errors instead of infinite loops, don't complain now 😝😝😝

**Comment by theashyster** [comment]
@IvanSanchez Damn, I have to choose my expectations more wisely next time.

**Comment by theashyster** [comment]
@IvanSanchez Is this implementation going to stay the way it is with the error thrown continuously?

**Comment by IvanSanchez** [comment]
What would be the alternative? I can't see a good one right now :-|

**Comment by theashyster** [comment]
I played with the example a bit and looks like the error is thrown in an infinite loop basically when you do a drag and release that causes the inertia to kick in. In other words when I changed the example a bit and added `inertia: false` to the map options I was not able to reproduce the infinite loop of errors. Maybe that helps in finding some ways to prevent this?

**Comment by zhen1991** [comment]
Hi!
I had the same problem like u guys 
Uncaught Error: Attempted to load an infinite number of tiles
when using proj4leaflet js.
my code is look like that

<dom-module id="my-view2">
<link rel="import" href="../bower_components/polymer/polymer-element.html">
<link rel="import" href="shared-styles.html">
<link rel="import" href="../bower_components/shadycss/apply-shim.html">
<link type="css" rel="import" href="../leaflet-1.3.1/dist/leaflet.css" />
<link rel="stylesheet" href="https://unpkg.com/leaflet@1.3.1/dist/leaflet.css" />
<script src="https://unpkg.com/leaflet@1.3.1/dist/leaflet.js"></script>
<script src="//ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
<script type="text/javascript" src="http://maps.stamen.com/js/tile.stamen.js?v1.3.0"></script>
<script src="../leaflet-1.3.1/dist/leaflet.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/proj4js/2.4.4/proj4.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/proj4leaflet/1.0.2/proj4leaflet.js"></script>
<script type="text/javascript" src="https://cdn.rawgit.com/jieter/Leaflet.encoded/68e78190/Polyline.encoded.js"
crossorigin=""></script>
  <template>
    <style include="shared-styles">
      :host {
        display: block;

        padding: 10px;
      }
      #mapid { position:absolute; top:0; bottom:0; width:100%;  max-width: none;
   max-height: none; display: inline}
   
    </style>
<div id="mapid"></div>
  </template>

  <script>
    class MyView2 extends Polymer.Element {
      static get is() { return 'my-view2'; }
      connectedCallback(){
        super.connectedCallback();    
     var crs = new L.Proj.CRS('EPSG:3006',
  '+proj=utm +zone=33 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs',
  {
    resolutions: [
      8192, 4096, 2048, 1024, 512, 256, 128
    ],
    origin: [218128.7031, 6126002.9379]
  });
     console.log(crs);
      var map = new L.Map(this.$.mapid, {
      crs: crs,
      center: new L.LatLng(55.729043, 13.226852), // Universal Lat/Lng
        zoom: 9,
      continuousWorld: true,
      });
      console.log(map);
    L.tileLayer("http://map0{s}.eniro.no/geowebcache/service/tms1.0.0/map/{z}/{x}/{y}.png", {
     maxZoom: crs.options.resolutions.length,
     minZoom: 0,
    continuousWorld: true,
     "tms":true, "subdomains":"1234", "attribution": "Eniro Maps" 
    }).addTo(map);

    console.log(map,"finally");
   /*

**Comment by zhen1991** [comment]
Hi!
I had the same problem like u guys
Uncaught Error: Attempted to load an infinite number of tiles
when using proj4leaflet js.

**Comment by IvanSanchez** [comment]
Can you publish a fiddle/codepen/[plunkr](https://leafletjs.com/edit) that displays this behaviour? I mean, something that reproduces the issue while providing a non-infinite `latlng`.

**Comment by MappingSupport** [comment]
Greater Seattle area was hit today with the second windstorm in less than a week.  Assuming the power is on in the morning I will see if I can produce a short demo.

**Comment by IvanSanchez** [comment]
I think the current approach (throw an error from #6416) is the right one - no better ones have been suggested :-|
